pyOlog/ophyd_tools.py

```diff
@@ -445,11 +445,14 @@ def _print_pos(positioners, file=sys.stdout):
         if pos is None:
             continue
         if v is not None:
-            try:
-                prec = p.precision
-            except (AttributeError, DisconnectedError):
-                prec = FMT_PREC
-            value = np.round(v, decimals=prec)
+           if isinstance(v,'float'):
+                try:
+                    prec = p.precision
+                except (AttributeError, DisconnectedError):
+                    prec = FMT_PREC
+                value = np.round(v, decimals=prec)
```

Comment:
This might fail on some weird edge cases due to duck-typing.

I suggest

```python
try:
    value = np.round(v, decimals=prec)
except TypeError:
    value = v
```

Reply:
Thanks @tacaswell , I was not aware of the weird edge cases. I will make the change you suggest.